odl/operator/operator.py

```diff
@@ -206,29 +395,64 @@ class Operator(with_metaclass(_OperatorMeta, object)):
     implementation of the respective other is provided.
     """
 
+    def __new__(cls, *args, **kwargs):
+        """Create a new instance."""
+        instance = super().__new__(cls)
+
+        call_has_out, call_out_optional, _ = _dispatch_call_args(cls)
+        instance._call_has_out = call_has_out
+        instance._call_out_optional = call_out_optional
+        if not call_has_out:
+            # Out-of-place _call
+            instance._call_in_place = partial(_default_call_in_place,
+                                              instance)
+            instance._call_out_of_place = instance._call
+        elif call_out_optional:
+            # Dual-use _call
+            instance._call_in_place = instance._call
+            instance._call_out_of_place = instance._call
+        else:
+            # In-place only _call
+            instance._call_in_place = instance._call
+            instance._call_out_of_place = partial(_default_call_out_of_place,
+                                                  instance)
+
+        return instance
+
     def __init__(self, domain, range, linear=False):
         """Initialize a new instance.
 
         Parameters
         ----------
-        dom : `Set`
+        domain : `Set`
             The domain of this operator, i.e., the set of elements to
             which this operator can be applied
-
-        ran : `Set`
+        range : `Set`
             The range of this operator, i.e., the set this operator
             maps to
+        linear : bool
+            If `True`, the operator is considered as linear. In this
+            case, `domain` and `range` have to be instances of
+            `LinearSpace`, `RealNumbers` or `ComplexNumbers`.
         """
         if not isinstance(domain, Set):
-            raise TypeError('domain {!r} not a `Set` instance.'.format(domain))
+            raise TypeError('domain {!r} is not a `Set` instance.'
+                            ''.format(domain))
         if not isinstance(range, Set):
-            raise TypeError('range {!r} not a `Set` instance.'.format(range))
+            raise TypeError('range {!r} is not a `Set` instance.'
+                            ''.format(range))
 
         self._domain = domain
         self._range = range
         self._is_linear = bool(linear)
         self._is_functional = isinstance(range, Field)
 
+        # Mandatory out makees no sense for functionals
+        if (self._is_functional and self._call_has_out and
+                not self._call_out_optional):
+            raise ValueError('mandatory `out` parameter not allowed for '
```

Comment:
Now I remember. We need optional `out` as a way to support vectorized functions, which are exactly this: functionals (in the duck-typing sense) with optional `out`.